behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
+    const {isLoading: isLoadingArtists, usersData} = useUsers("all_artists", [], userFilter)
+
+    const serviceFilter = {serviceActive: true}
+    const {isLoading: isLoadingServices, servicesData} = useServices("active_services", [], serviceFilter, true)
+
+    // variables set up for suggestions
+    const [fullSuggestions, setFullSuggestions] = useState([]);
+    const [loadedFullSuggestions, setLoadedFullSuggestions] = useState(false);
+    const [filteredSuggestions, setFilteredSuggestions] = useState([]);
+    const [selectedSuggestions, setSelectedSuggestions] = useState(false);
+    const [maxSuggestion, setMaxSuggestion] = useState(10);
+
+    /**
+     * this function is used when the user wants to submit the value in the search bar, either by enter key or a button
+     * Alter this function with whatever data manipulation is needed to be done with the input value
+     */
+    const handleButtonClickOrSubmit = (event) => {
+        event ? event.preventDefault() : null; // This line is important as it prevents the automatic submit for forms which reloads the page
+        if (searchType === "services") {
+            navigate("/" + UrlBasePath.SERVICES + "/?search=" + inputValue)
+        } else if (searchType === "artists") {
+            navigate("/" + UrlBasePath.ARTISTS + "/?search=" + inputValue);
+        }
+        setFilteredSuggestions([]);
+    };
+
+    const onClickInput = () => {
+        // if it has any input, display suggestions
+        if (inputValue !== "") {
+            // Filter suggestions based on input value
+            const filteredSuggestions = fullSuggestions.filter(suggestion => {
+                // filter suggestions based on main or sub
+                const matchMain = suggestion.main.toLowerCase().includes(inputValue.toLowerCase());
+                const matchSub = suggestion.sub ? suggestion.sub.toLowerCase().includes(inputValue.toLowerCase()) : false;
+                return matchMain || matchSub;
+            });
+
+            setFilteredSuggestions(filteredSuggestions);
+        } else {
+            setFilteredSuggestions([]);
+        }
+    }
+
+    // handler for when user clicks OFF the search bar input (hide the suggestions)
+    const onBurInput = (event) => {
+        // get the element that user has clicked on
+        const clickTarget = event.relatedTarget;
+
+        // check if it is a suggestion (if it is a suggestion do NOT clear the list because then the suggestion
+        // will disappear before click can be registered
+        if (clickTarget && clickTarget.getAttribute("name") === "suggestion-list-item") {
+            return;
+        }
+        // in all other cases, empty/hide suggestions
+        setFilteredSuggestions([]);
+    }
+
+    // this function updates the state when input changes
+    const handleInputChange = (event) => {
+        setInputValue(event.target.value);
+    };
+
+    // update suggestions on input change
+    useEffect(() => {
```

Comment:
In general, it's best to use useEffect sparingly (https://react.dev/learn/you-might-not-need-an-effect).

In this case, inputValue is defined in this component, so we should have full control over it and I don't thinkkk we need a useEffect to achieve this logic